### PR TITLE
Rating: Add support for single selection and empty symbols

### LINKED ISSUE #4: Rating component
Create an input for rating that can both show the rating of an item and let the user choose a value e.g. when leaving feedback
image

### PR DESCRIPTION
**Pull request by simeonoff**
Closes #4

### CODE REVIEW

File: src/components/rating/rating.ts [hunk 1]
```diff
@@ -3,21 +3,22 @@ import {
...
-import { styleMap } from 'lit/directives/style-map.js';
 import { ifDefined } from 'lit/directives/if-defined.js';
+import { styleMap } from 'lit/directives/style-map.js';
+import { themes } from '../../theming/theming-decorator.js';
 import { watch } from '../common/decorators/watch.js';
 import { Constructor } from '../common/mixins/constructor.js';
 import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
 import { SizableMixin } from '../common/mixins/sizable.js';
+import { clamp } from '../common/util.js';
+import IgcRatingSymbolComponent from './rating-symbol';
```
Comment (rkaraivanov) [inline src/components/rating/rating.ts:17]:
```suggestion
import IgcRatingSymbolComponent from './rating-symbol.js';
// OR
import type { IgcRatingSymbolComponent  } from './rating-symbol';
```

File: src/components/rating/rating.ts [hunk 2]
```diff
@@ -126,6 +131,10 @@ export default class IgcRatingComponent extends SizableMixin(
   @property({ type: Boolean, reflect: true })
   public readonly = false;
 
+  /** Toggles single selection visual mode. */
+  @property({ type: Boolean })
```
Comment (rkaraivanov) [inline src/components/rating/rating.ts:135]: I suppose this should be a reflected property 
```suggestion
  @property({ type: Boolean, reflect: true })
```

**Review by rkaraivanov — CHANGES_REQUESTED** [review]
LGTM with a couple of nitpicks

File: src/components/rating/rating.ts (hunk 2 above)
Comment (simeonoff) [reply src/components/rating/rating.ts:135]: This is not needed for styling, so no.

**Review by rkaraivanov — APPROVED** [review]

**Pull request merged**